clip/download.py

```diff
@@ -0,0 +1,51 @@
+"""Taken from https://github.com/ml-explore/mlx-examples/issues/155#issuecomment-1868120534"""
+import argparse
+import os
+
+import requests
+from tqdm import tqdm
+
```

Comment:
I think it's cleaner to use `snapshot_download` and we already have a dependency for it. Check this [example](https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/utils.py#L57-L82).

Concretely you could merge the download and convert scripts very easily.
1. Take a HF repo and snapshot download the files you need. This will cache the files and give you a path.
2. Load what you need from the path and convert to MLX

Reply:
This is a great idea. I will implement something similar to the example.